src/components/other/parallax-effect/ParallaxEffect.ts

```diff
@@ -0,0 +1,138 @@
+import { gsap } from 'gsap';
+import { ScrollTrigger } from 'gsap/ScrollTrigger';
+import { Component, dataParam, Logger, register } from 'ovee.js';
+
+gsap.registerPlugin(ScrollTrigger);
+
+export interface ParallaxTweenVars {
+	x?: gsap.TweenValue;
+	y?: gsap.TweenValue;
+	xPercent?: gsap.TweenValue;
+	yPercent?: gsap.TweenValue;
+}
+
+export interface ParallaxConfig extends ScrollTrigger.StaticVars {
+	trigger: gsap.DOMTarget;
+	disableOnMobile: boolean;
+	disableOnTablet: boolean;
+	tweenVars: ParallaxTweenVars;
+}
+
+const logger = new Logger('ParallaxEffect');
+
+@register('parallax-effect')
+export class ParallaxEffect extends Component {
+	tl: gsap.core.Timeline;
+	st: ScrollTrigger;
+
+	defaultParallaxConfig: ParallaxConfig = {
+		trigger: this.$element,
+		disableOnMobile: true,
+		disableOnTablet: true,
+		tweenVars: {
+			y: 100,
+		},
+	};
+
+	@dataParam('parallaxConfig')
+	_parallaxConfig = '';
+
+	get parallaxConfig(): ParallaxConfig {
+		if (!this._parallaxConfig) {
+			return this.defaultParallaxConfig;
+		}
+
+		let jsonConfig: ParallaxConfig = {
+			...this.defaultParallaxConfig,
+		};
+
+		try {
+			jsonConfig = JSON.parse(this._parallaxConfig);
+		} catch (e) {
+			logger.error('Invalid JSON Config:', this._parallaxConfig);
+
+			return {
+				...this.defaultParallaxConfig,
+			};
+		}
+
+		return {
+			...this.defaultParallaxConfig,
+			...jsonConfig,
+		};
+	}
+
+	get desktopBreakpoint() {
+		return '(min-width: 1200px)';
+	}
+
+	get tabletBreakpoint() {
+		return '(max-width: 1199px)';
+	}
+
+	get mobileBreakpoint() {
+		return '(max-width: 767px), (max-width: 1023px) and (orientation: landscape)';
+	}
```

Comment:
Jedyne co mi przyszło do głowy, to żeby przerobić te gettery na:
``` 
type Breakpoint = 'desktop' | 'tablet' | 'mobile';

const breakpoints: Record<Breakpoint, string> = {
    desktop: '(min-width: 1200px)',
    tablet: '(max-width: 1199px)',
    mobile: '(max-width: 767px), (max-width: 1023px) and (orientation: landscape)'
};

...

breakpoints.tablet
```
lub 
```
enum Breakpoint {
    Desktop: '(min-width: 1200px)',
    Tablet: '(max-width: 1199px)',
    Mobile: '(max-width: 767px), (max-width: 1023px) and (orientation: landscape)'
}

...

Breakpoint.Tablet
```

ale nie wiem czy to jest potrzebne, sam oceń 😄